openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSL.java

```diff
@@ -699,4 +702,68 @@ public static native void setCertificateBio(
      * Free x509 chain ({@code STACK_OF(X509)} pointer).
      */
     public static native void freeX509Chain(long x509Chain);
+    
+    // Extracted from https://github.com/openssl/openssl/blob/master/include/openssl/tls1.h
+    
+    /**
+     * This is the default (initial) status type value for {@link SSLEngine}s.
+     * 
+     * @see #setStatusType(long, int)
+     */
+    public static final int TLSEXT_STATUSTYPE_HOSTNAME = 0;
+    
+    /**
+     * Enables OCSP stapling on a {@link SSLEngine}.
+     * 
+     * https://www.openssl.org/docs/man1.0.2/ssl/SSL_set_tlsext_status_type.html
+     * 
+     * @see #setStatusType(long, int)
+     */
+    public static final int TLSEXT_STATUSTYPE_OCSP = 1;
+    
+    /**
+     * Returns {@code true} if OCSP stapling is supported.
+     */
+    public static native boolean isOcspAvailable();
+    
+    /**
+     * Enables or disables OCSP stapling on the given {@link SSLEngine}.
+     *
+     * https://www.openssl.org/docs/man1.0.2/ssl/SSL_set_tlsext_status_type.html
+     * 
+     * @see #setStatusType(long, int)
+     */
+    public static boolean setOcspEnabled(long ssl, boolean enabled) {
+        return setStatusType(ssl, enabled ? TLSEXT_STATUSTYPE_OCSP : TLSEXT_STATUSTYPE_HOSTNAME);
+    }
+    
+    /**
+     * Sets the status type on the given {@link SSLEngine}.
+     * 
+     * @see #TLSEXT_STATUSTYPE_HOSTNAME
+     * @see #TLSEXT_STATUSTYPE_OCSP
+     */
+    public static native boolean setStatusType(long ssl, int type);
```

Comment:
can we make `setStatusType` more descriptive? Consider something like `setTlsExtStatusType`.